src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/StaffConfigManager.kt

```diff
@@ -58,6 +64,86 @@ class StaffConfigManager(
     }
   }
 
+  fun patchStaffConfiguration(
+    prisonCode: String,
+    staffId: Long,
+    request: PatchStaffConfigRequest,
+  ) {
+    val staffConfig = staffConfigRepository.findByStaffId(staffId) ?: throw EntityNotFoundException("Staff config not found")
+    if (!request.jobClassification.isPresent || request.jobClassification.get() != null) {
+      staffConfig.patch(request)
+    }
+
+    if (request.deactivateActiveAllocations.orElse(false) ||
```

Comment:
I'd switch the order if this deactivate and the removal of the job classification. We have to call another API to remove the KW job responsibility which could fail leaving the staff member with no allocations but still being a key worker